tests/unittests/Foundation/NSThreadTests.m

```diff
@@ -42,6 +42,12 @@ - (void)countingloop {
 static const double sExpectedLowThreadPriority = 0;
 static const double sExpectedMidThreadPriority = .5;
 
+// A thread never has a set priority. The system can always change it, in fact it will most likely change the priority.
+// Even still, the thread priority should be close enough to the actual requested value so we need to make sure it's sufficiently close.
+// The largest difference between thread priorities in Windows should be .07 units in difference as Windows pigeon holes priorities into
+// certain values.
+static const double sMarginOfError = .07;
```

Comment:
Yes, the system can essentially do whatever it wants with the thread priority. From the reference platform's documentation: "...but because the priority is determined by the kernel, there is no guarantee what this value actually will be." That said, I took the larger of the two values that I've seen the priority change by. In this case, it's Windows since the system priorities are bucketized to particular values that I've mapped to more expected values. If the priority changes by more than .07 then that indicates the priority of the thread was changed by more than the size of one of these buckets which means the priority was not preserved.

I've seen the OSX values change significantly less on an average basis however they change more often. Usually by a couple decimals in either direction. Seems like the reference platform is very hands on with thread priorities.

---
In reply to: [73447706](https://github.com/Microsoft/WinObjC/pull/802#discussion_r73447706) [](ancestors = 73447706)